### PR TITLE
get_edge_target_turf() now finds the proper turf when using diagonal directions

### PR DESCRIPTION
**Pull request by ChangelingRain**
hey

### CODE REVIEW

File: code/__HELPERS/unsorted.dm [hunk 1]
```diff
@@ -442,19 +442,23 @@ Turf and target are seperate in case you want to teleport some distance from a t
 	var/turf/target = locate(A.x, A.y, A.z)
 	if(!A || !target)
 		return 0
-		//since NORTHEAST == NORTH & EAST, etc, doing it this way allows for diagonal mass drivers in the future
+		//since NORTHEAST == NORTH|EAST, etc, doing it this way allows for diagonal mass drivers in the future
 		//and isn't really any more complicated
 
-		// Note diagonal directions won't usually be accurate
```
Comment (ChangelingRain) [inline code/__HELPERS/unsorted.dm]: diagonal directions were not considered at all, actually, so they'd always target EAST or WEST

File: code/game/machinery/camera/camera.dm [hunk 2]
```diff
@@ -324,19 +324,10 @@
 /atom/proc/auto_turn()
 	//Automatically turns based on nearby walls.
 	var/turf/closed/wall/T = null
-	for(var/i = 1, i <= 8; i += i)
+	for(var/i in GLOB.cardinals)
 		T = get_ranged_target_turf(src, i, 1)
 		if(istype(T))
-			//If someone knows a better way to do this, let me know. -Giacom
```
Comment (ChangelingRain) [inline code/game/machinery/camera/camera.dm]: @Giacom hey

File: code/game/machinery/camera/camera.dm [hunk 3]
```diff
@@ -324,19 +324,10 @@
 /atom/proc/auto_turn()
 	//Automatically turns based on nearby walls.
 	var/turf/closed/wall/T = null
-	for(var/i = 1, i <= 8; i += i)
+	for(var/i in GLOB.cardinal)
 		T = get_ranged_target_turf(src, i, 1)
 		if(istype(T))
-			//If someone knows a better way to do this, let me know. -Giacom
```
Comment (ChangelingRain) [inline code/game/machinery/camera/camera.dm]: @Giacom hey

File: code/game/machinery/camera/camera.dm [hunk 4]
```diff
@@ -324,19 +324,10 @@
...
-			switch(i)
-				if(NORTH)
-					src.setDir(SOUTH)
-				if(SOUTH)
-					src.setDir(NORTH)
-				if(WEST)
-					src.setDir(EAST)
-				if(EAST)
-					src.setDir(WEST)
+			setDir(turn(i, 180)
```
Comment (RemieRichards) [inline code/game/machinery/camera/camera.dm]: missing a parenthesis is clearly a better way yup.

Comment (ChangelingRain) [inline code/game/machinery/camera/camera.dm]: increases performance by INFINITY if you can never run the server!

**Review by Cyberboss — APPROVED** [review]

File: code/game/machinery/camera/camera.dm [hunk 5]
```diff
@@ -324,19 +324,10 @@
 /atom/proc/auto_turn()
 	//Automatically turns based on nearby walls.
 	var/turf/closed/wall/T = null
-	for(var/i = 1, i <= 8; i += i)
+	for(var/i in GLOB.cardinal)
```
Comment (ChangelingRain) [inline code/game/machinery/camera/camera.dm]: good job cyberboss

**Pull request merged**